pandas/indexes/base.py

```diff
@@ -199,14 +199,25 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
                             data = np.array(data, copy=copy, dtype=dtype)
                         elif inferred in ['floating', 'mixed-integer-float']:
 
-                            # if we are actually all equal to integers
+                            # If we are actually all equal to integers,
                             # then coerce to integer
-                            from .numeric import Int64Index, Float64Index
+                            from .numeric import (Int64Index, UInt64Index,
+                                                  Float64Index)
                             try:
-                                res = data.astype('i8')
+                                res = data.astype('i8', copy=False)
                                 if (res == data).all():
                                     return Int64Index(res, copy=copy,
                                                       name=name)
+                            except (OverflowError, TypeError, ValueError):
+                                pass
```

Comment:
I don't believe so.  `TypeError` and `ValueError` were being guarded against for `Int64Index` initialization before.  I see no reason to change that.  The other option is to do something like this:
~~~python
try:
   try:
       Int64Index(...)
   except OverflowError:
       UInt64Index(...)
except (TypeError, ValueError):
   pass
~~~

However, I am more in favor of my method at this point since it is slightly more modular.